sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -946,7 +946,21 @@ class Dataset[T] private[sql](
   }
 
   /**
-   * Inner equi-join with another `DataFrame` using the given columns.
+   * (Java-specific) Inner equi-join with another `DataFrame` using the given columns. See the
+   * Scala-specific overload for more details.
+   *
+   * @param right Right side of the join operation.
+   * @param usingColumns Names of the columns to join on. This columns must exist on both sides.
+   *
+   * @group untypedrel
+   * @since 3.4.0
+   */
+  def 4oin(right: Dataset[_], usingColumns: Array[String]): DataFrame = {
```

Comment:
hmm.... `4oin` -> `join` ?

Reply:
Oh man that's embarrassing, I blame vim.